pkg/alert/prometheusrule_test.go

```diff
@@ -1,207 +1,119 @@
-package alert_test
+package alert
 
 import (
-	"github.com/golang/mock/gomock"
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
 	"context"
-
-	// tested package
+	"testing"
 
 	"github.com/openshift/route-monitor-operator/api/v1alpha1"
-	"github.com/openshift/route-monitor-operator/pkg/alert"
-	consterror "github.com/openshift/route-monitor-operator/pkg/consts/test/error"
-	constinit "github.com/openshift/route-monitor-operator/pkg/consts/test/init"
-
-	clientmocks "github.com/openshift/route-monitor-operator/pkg/util/test/generated/mocks/client"
-	utilmock "github.com/openshift/route-monitor-operator/pkg/util/test/generated/mocks/reconcile"
-	testhelper "github.com/openshift/route-monitor-operator/pkg/util/test/helper"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-type ResourceComparerMockHelper struct {
-	CalledTimes int
-	ReturnValue bool
-}
-
-var _ = Describe("CR Deployment Handling", func() {
-	var (
-		ctx                  context.Context
-		mockClient           *clientmocks.MockClient
-		mockCtrl             *gomock.Controller
-		mockResourceComparer *utilmock.MockResourceComparerInterface
-
-		deepEqual ResourceComparerMockHelper
-		get       testhelper.MockHelper
-		create    testhelper.MockHelper
-		update    testhelper.MockHelper
-		delete    testhelper.MockHelper
-
-		prometheusRuleRef v1alpha1.NamespacedName
-		prometheusRule    monitoringv1.PrometheusRule
-		pr                alert.PrometheusRule
-		err               error
+func TestTemplateAndUpdatePrometheusRuleDeployment(t *testing.T) {
+	const (
+		url                     = "example.com"
+		percent                 = "99"
+		prometheusRuleName      = "test-name"
+		prometheusRuleNamespace = "test-namespace"
 	)
-	BeforeEach(func() {
-		ctx = constinit.Context
-		mockCtrl = gomock.NewController(GinkgoT())
-		mockClient = clientmocks.NewMockClient(mockCtrl)
-		mockResourceComparer = utilmock.NewMockResourceComparerInterface(mockCtrl)
 
-		deepEqual = ResourceComparerMockHelper{}
-		get = testhelper.MockHelper{}
-		create = testhelper.MockHelper{}
-		update = testhelper.MockHelper{}
-		delete = testhelper.MockHelper{}
-
-		prometheusRuleRef = v1alpha1.NamespacedName{}
-		prometheusRule = monitoringv1.PrometheusRule{}
+	var namespacedName = types.NamespacedName{
```

Comment:
+1 to simplify unit tests. However, the rest of the code base uses BDD tests with Gomega assertions. Why do you think this specific file would benefit from table tests using `assert.` assertions? Should we use Ginkgo table tests instead?